vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/LinkManager.java

```diff
@@ -89,182 +36,73 @@
     @Autowired
     private IStorageEngine storage;
 
-    @Autowired
-    private IModuleManager moduleManager;
-
-    @Autowired
-    private IModuleLinkFactory moduleLinkFactory;
-
-    @Autowired
-    private IModuleLinkDisplayFactory moduleLinkDisplayFactory;
-
-    @Autowired
-    private ModuleLinkRepository moduleLinkRepo;
-
-    @Autowired
-    private ModuleLinkDisplayRepository moduleLinkDisplayRepo;
-
-    /*
-     * ==== Space links ====
-     */
+    public ILinkDisplay createLink(String title, String id, float positionX, float positionY,
+            int rotation, String linkedId, String linkLabel, DisplayType displayType, byte[] linkImage,
+            String imageFilename) throws SpaceDoesNotExistException,ImageCouldNotBeStoredException, SpaceDoesNotExistException{
 
-    /*
-     * (non-Javadoc)
-     * 
-     * @see
-     * edu.asu.diging.vspace.core.services.impl.ILinkManager#createSpaceLink(java.
-     * lang.String, edu.asu.diging.vspace.core.model.ISpace, float, float, int,
-     * java.lang.String, java.lang.String,
-     * edu.asu.diging.vspace.core.model.display.DisplayType)
-     */
-    @Override
-    public ISpaceLinkDisplay createSpaceLink(String title, ISpace source, float positionX, float positionY,
-            int rotation, String linkedSpaceId, String spaceLinkLabel, DisplayType displayType, byte[] linkImage,
-            String imageFilename) throws ImageCouldNotBeStoredException, SpaceDoesNotExistException {
-        // we need this to fully load the space
-        source = spaceManager.getSpace(source.getId());
-        if (source == null) {
-            throw new SpaceDoesNotExistException();
-        }
+        IVSpaceElement target = getTarget(linkedId);
+        ILink link = (ILink) createLinkObject(title, id, target, linkLabel); 
+        ILinkDisplay displayLink = createLinkDisplay(link);
+        setDisplayProperties(displayLink, positionX, positionY, rotation, displayType, linkImage, imageFilename);
+        return saveLinkAndDisplay(link, displayLink);
```

Comment:
if you add an abstract method `getLinkRepo` and `getDisplayRepo`, then you don't need saveLinkAndDisplay and you might be able to get rid of other methods as well.

Reply:
here we are creating and saving a new link and display link. getLinkRepo and getDisplayRepo are used in update methods. getting rid of saveLinkAndDisplay from create method though.